### PR TITLE
Remove any references to OCP in the helm charts and upstream manifests

### LINKED ISSUE #2130: Remove the root manifests directory and OCP/downstream helm charts
The root manifests are no longer needed now that a new repository was created to house the downstream OCP builds, and this repository was decoupled from the OCP release cycle.

We can similarly investigate removing the OCP-related helm charts entirely from upstream.

### PR DESCRIPTION
**Pull request by timflannagan**
Fixes #2130 

Continue to decouple the (now) upstream repository from the OCP platform after #2153 landed.

- Removes the OCP helm chart templates and the rendered manifests.
- Removes any OCP-related helm chart conditionals.
- Removes any references in the Makefile and scripts/ bash scripts for handling OCP helm charts and downstream manifests.

A couple of things to note:
- We may want to deflate the deploy/upstream -> deploy in this PR, or a future PR.
- I kept the current PrometheusRules and ServiceMonitor helm charts as those appear to be agnostic of OCP and live as upstream CRDs in the various Prometheus-related repositories. Those helm charts are still gated by the OCP platform conditional though.
- Do we want to continue naming the manifests by the CVO run levels, or remove that entirely?

### CODE REVIEW

**Comment by timflannagan** [comment]
There's a couple of outstanding questions in my PR description that I'd like to get addressed before merging this.

/hold

File: deploy/chart/templates/0000_50_olm_08-catalog-operator.deployment.yaml [hunk 1]
```diff
@@ -64,26 +61,10 @@ spec:
               path: /healthz
               port: {{ .Values.catalog.service.internalPort }}
           terminationMessagePolicy: FallbackToLogsOnError
-          env:
```
Comment (timflannagan) [inline deploy/chart/templates/0000_50_olm_08-catalog-operator.deployment.yaml:67]: A quick note as I need to double-check against the current master branch, but this appeared to be problematic in the context of the upstream manifests as we'd essentially be rendering an empty `env` map as the only key would be that RELEASE_VERSION variable, but that conditional check would evaluate to false.

File: Makefile [hunk 2]
```diff
@@ -29,12 +29,12 @@ GO := GO111MODULE=on GOFLAGS="$(MOD_FLAGS)" go
...
-# ART builds are performed in dist-git, with content (but not commits) copied 
+# ART builds are performed in dist-git, with content (but not commits) copied
 # from the source repo. Thus at build time if your code is inspecting the local
-# git repo it is getting unrelated commits and tags from the dist-git repo, 
+# git repo it is getting unrelated commits and tags from the dist-git repo,
 # not the source repo.
-# For ART image builds, SOURCE_GIT_COMMIT, SOURCE_GIT_TAG, SOURCE_DATE_EPOCH 
-# variables are inserted in Dockerfile to enable recovering the original git 
+# For ART image builds, SOURCE_GIT_COMMIT, SOURCE_GIT_TAG, SOURCE_DATE_EPOCH
+# variables are inserted in Dockerfile to enable recovering the original git
```
[the remark this answers is not in the document]
Comment (benluddy) [reply Makefile]: No, I don't think there's any reason to keep that around.

**Pull request merged**